Kudu.Core/Infrastructure/StringExtensions.cs

```diff
@@ -28,5 +29,27 @@ public static string EscapeHashCharacter(this string str)
         {
             return str.Replace("#", Uri.EscapeDataString("#"));
         }
+
+        // Returns a string you can safely use as valid Windows or Linux file name.
+        // Does not check for reserved words.
+        public static string ToSafeFileName(this string str)
+        {
+            foreach (char c in Path.GetInvalidFileNameChars())
+            {
+                str = str.Trim()
+                         .Replace(c, '-')
+                         .Replace(' ', '-');
```

Comment:
Fixed in https://github.com/projectkudu/kudu/pull/2040/commits/d1e78434b9aa38f708239990b1f87bd8081816b3